include/boost/geometry/policies/relate/intersection_policy.hpp

```diff
@@ -0,0 +1,215 @@
+    static inline return_type segments_crosses(side_info const& sides,
+                                        SegmentIntersectionInfo const& sinfo,
+                                        Segment1 const& s1, Segment2 const& s2)
+    {
+        return return_type
+            (
+                pts_policy::segments_crosses(sides, sinfo, s1, s2),
+                dir_policy::segments_crosses(sides, sinfo, s1, s2)
+            );
+    }
+
+    template <typename Segment1, typename Segment2, typename Ratio>
+    static inline return_type segments_collinear(
+                                        Segment1 const& segment1,
+                                        Segment2 const& segment2,
+                                        bool opposite,
+                                        int pa1, int pa2, int pb1, int pb2,
+                                        Ratio const& ra1, Ratio const& ra2,
+                                        Ratio const& rb1, Ratio const& rb2)
+    {
+        return return_type
+            (
+                pts_policy::segments_collinear(segment1, segment2,
+                                               opposite,
+                                               pa1, pa2, pb1, pb2,
+                                               ra1, ra2, rb1, rb2),
+                dir_policy::segments_collinear(segment1, segment2,
+                                               opposite,
+                                               pa1, pa2, pb1, pb2,
+                                               ra1, ra2, rb1, rb2)
+            );
+    }
+
+    template <typename Segment>
+    static inline return_type degenerate(Segment const& segment,
+                                         bool a_degenerate)
+    {
+        return return_type
+            (
+                pts_policy::degenerate(segment, a_degenerate),
+                dir_policy::degenerate(segment, a_degenerate)
+            );
+    }
+
+    template <typename Segment, typename Ratio>
+    static inline return_type one_degenerate(Segment const& segment,
+                                             Ratio const& ratio,
+                                             bool a_degenerate)
+    {
+        return return_type
+            (
+                pts_policy::one_degenerate(segment, ratio, a_degenerate),
+                dir_policy::one_degenerate(segment, ratio, a_degenerate)
+            );
+    }
+
+    static inline return_type disjoint()
+    {
+        return return_type
+            (
+                pts_policy::disjoint(),
+                dir_policy::disjoint()
+            );
+    }
+
+    static inline return_type error(std::string const& msg)
+    {
+        return return_type
+            (
+                pts_policy::error(msg),
+                dir_policy::error(msg)
+            );
+    }
+};
+
+
+// "tupled" to return intersection results together.
+// NOTE: unused, left for backward compatibility
```

Comment:
Is it unused? I think it's still used.
If not used at all, I've no objection to remove it.

Reply:
I removed it.